demos/coreHTTP/http_demo_mutual_auth.c

```diff
@@ -0,0 +1,444 @@
+/*
+ * FreeRTOS V202011.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ */
+
+/*
+ * Demo for showing use of the HTTP API using a mutually-authenticated network
+ * connection.
+ *
+ * The example shown below uses HTTP APIs to first create a
+ * mutually-authenticated network connection with an HTTP server, and then send
+ * a POST request containing a simple message. This example is single threaded
+ * and uses statically allocated memory. It uses QoS1 for sending and receiving
+ * messages from the server.
+ *
+ * A mutually-authenticated TLS connection is used to connect to the HTTP server
+ * in this example. Define democonfigAWS_IOT_ENDPOINT and democonfigROOT_CA_PEM
+ * in http_demo_mutual_auth_config.h, and define the client private key and
+ * certificate in aws_clientcredential_keys.h, to establish a mutually
+ * authenticated connection.
+ */
+
+/**
+ * @file http_demo_mutual_auth.c
+ * @brief Demonstrates usage of the HTTP library.
+ */
+/* Standard includes. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Demo Specific configs. */
+#include "http_demo_mutual_auth_config.h"
+
+/* Include common demo header. */
+#include "aws_demo.h"
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_secure_sockets.h"
+
+/* Common HTTP demo utilities. */
+#include "http_demo_utils.h"
+
+/* HTTP API header. */
+#include "core_http_client.h"
+
+/* Include header for client credentials. */
+#include "aws_clientcredential_keys.h"
```

Comment:
What do you need from this header in this file?

Reply:
Should you include aws_clientcredential.h for the iot endpoint?

Reply:
Nothing really -- removed it, thanks.